generators/java-v2/sdk/src/SdkGeneratorContext.ts

```diff
@@ -26,4 +38,215 @@ export class SdkGeneratorContext extends AbstractJavaGeneratorContext<SdkCustomC
+    }
+
+    public getPackageLayout(): string {
+        return this.customConfig?.["package-layout"] ?? "nested";
+    }
+
+    public getPaginationClassName(): string {
+        return "SyncPagingIterable";
+    }
+
+    public getOkHttpClientClassReference(): java.ClassReference {
+        return java.classReference({
+            name: "OkHttpClient",
+            packageName: "okhttp3"
+        });
+    }
+
+    public getRequestOptionsClassReference(): java.ClassReference {
+        return java.classReference({
+            name: this.getRequestOptionsClassName(),
+            packageName: this.getCorePackageName()
+        });
+    }
+
+    public getRequestOptionsClassName(): string {
+        return "RequestOptions";
+    }
+
+    public getApiExceptionClassReference(): java.ClassReference {
+        return java.classReference({
+            name: this.getApiExceptionClassName(),
+            packageName: this.getCorePackageName()
+        });
+    }
+
+    public getApiExceptionClassName(): string {
+        return this.customConfig?.["base-api-exception-class-name"] ?? `${this.getBaseNamePrefix()}ApiException`;
+    }
+
+    public getEnvironmentClassReference(): java.ClassReference {
+        return java.classReference({
+            name: this.getEnvironmentClassName(),
+            packageName: this.getCorePackageName()
+        });
+    }
+
+    public getCorePackageName(): string {
+        const tokens = this.getPackagePrefixTokens();
+        tokens.push("core");
+        return this.joinPackageTokens(tokens);
+    }
+
+    public getEnvironmentClassName(): string {
+        return "Environment";
+    }
+
+    public getRootClientClassReference(): java.ClassReference {
+        return java.classReference({
+            name: this.getRootClientClassName(),
+            packageName: this.getRootPackageName()
+        });
+    }
+
+    public getRootPackageName(): string {
+        const tokens = this.getPackagePrefixTokens();
+        return this.joinPackageTokens(tokens);
+    }
+
+    public getRootClientClassName(): string {
+        return this.customConfig?.["client-class-name"] ?? `${this.getBaseNamePrefix()}Client`;
+    }
+
+    private joinPackageTokens(tokens: string[]): string {
+        const sanitizedTokens = tokens.map((token) => {
+            return this.startsWithNumber(token) ? "_" + token : token;
+        });
+        return sanitizedTokens.join(".");
+    }
+
+    private startsWithNumber(token: string): boolean {
+        return /^\d/.test(token);
+    }
+
+    private getPackagePrefixTokens(): string[] {
+        if (this.customConfig?.["package-prefix"] != null) {
+            return this.customConfig["package-prefix"].split(".");
+        }
+        const prefix: string[] = [];
+        prefix.push("com");
+        prefix.push(...this.splitOnNonAlphaNumericChar(this.config.organization));
+        prefix.push(...this.splitOnNonAlphaNumericChar(this.getApiName()));
+        return prefix;
+    }
```

Comment:
Nice, great to add all this - we can follow-up to try and consolidate all these utilities between the dynamic snippets context and the SDK context. That's been a long running issue on my mind (but def out of scope from this change).